crates/bevy_ecs/src/resource/resources.rs

```diff
@@ -320,6 +320,29 @@ impl Resources {
         })
     }
 
+    pub fn take_global_only_resource<T: Resource>(&mut self) -> Option<T> {
```

Comment:
This name/impl definitely makes sense in the context of the current implementation, but it makes less sense from a user facing perspective. Ideally we could have `resources.take::<T>()`, which takes a global, and `resources.take_local::<T>(index)` which takes a local.